File: test/cpp/polyglot/regression/831.cpp
// Copyright 2015 Adam Grandquist

#include "./catch.hpp"
#include "./ReQL.hpp"

using namespace ReQL;

TEST_CASE("cpp 831 -- Empty batched_replaces_t constructed", "[cpp][ast]") {

  SECTION("test0") {
    std::map<std::string, Query> map0;

    std::string src1("deleted", 7);
    Query var1(src1);

    double num2(0);
    Query var2(num2);

    map0.insert({src1, var2});

    std::string src3("errors", 6);
    Query var3(src3);

    double num4(2);
    Query var4(num4);

    map0.insert({src3, var4});

    std::string src5("first_error", 11);
    Query var5(src5);

    std::string src6("Expected type OBJECT but found BOOL.", 36);
    Query var6(src6);

    map0.insert({src5, var6});

    std::string src7("inserted", 8);
    Query var7(src7);

    double num8(0);
    Query var8(num8);

    map0.insert({src7, var8});

    std::string src9("replaced", 8);
    Query var9(src9);

    double num10(0);
    Query var10(num10);

    map0.insert({src9, var10});

    std::string src11("skipped", 7);
    Query var11(src11);

    double num12(0);
    Query var12(num12);

    map0.insert({src11, var12});

    std::string src13("unchanged", 9);
    Query var13(src13);

    double num14(0);
    Query var14(num14);

    map0.insert({src13, var14});

    Query var0(map0);
  }
}

File: libReQLTests/polyglot/regression/831.m
// Copyright 2015 Adam Grandquist

#import <Cocoa/Cocoa.h>
#import <XCTest/XCTest.h>
#import <libReQL.h>

@interface Objc831EmptyBatchedReplacesTConstructed : XCTestCase

@end

@implementation Objc831EmptyBatchedReplacesTConstructed


- (void)test0 {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  NSMutableDictionary *var0 = [NSMutableDictionary dictionaryWithCapacity:7];
  NSString *var1 = @"deleted";
  NSNumber *var2 = @0;
  [var0 setObject:var2 forKey:var1];
  NSString *var3 = @"errors";
  NSNumber *var4 = @2;
  [var0 setObject:var4 forKey:var3];
  NSString *var5 = @"first_error";
  NSString *var6 = @"Expected type OBJECT but found BOOL.";
  [var0 setObject:var6 forKey:var5];
  NSString *var7 = @"inserted";
  NSNumber *var8 = @0;
  [var0 setObject:var8 forKey:var7];
  NSString *var9 = @"replaced";
  NSNumber *var10 = @0;
  [var0 setObject:var10 forKey:var9];
  NSString *var11 = @"skipped";
  NSNumber *var12 = @0;
  [var0 setObject:var12 forKey:var11];
  NSString *var13 = @"unchanged";
  NSNumber *var14 = @0;
  [var0 setObject:var14 forKey:var13];
  XCTAssert([var0 isEqualTo:var0]);
  [connection close];
}

@end
